> clang error:  argument '-Ofast' is deprecated; use '-O3 -ffast-math' for the same behavior, or '-O3' to enable only conforming optimizations [-Werror,-Wdeprecated-ofast]
make[2]: *** [CMakeFiles/mlx42.dir/build.make:98: CMakeFiles/mlx42.dir/src/mlx_exit.c.o] Error 1

given that they are compiling with `-Werror` this should be reported upstream so they can fix the problem -- they may not be aware that `-Ofast` implies `-ffast-math` which is one of the reasons this is depreciated.

--- a/pkgs/by-name/ml/mlx42/package.nix
+++ b/pkgs/by-name/ml/mlx42/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   cmake,
   glfw,
   enableShared ? !stdenv.hostPlatform.isStatic,
@@ -19,13 +20,23 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-/HCP6F7N+J97n4orlLxg/4agEoq4+rJdpeW/3q+DI1I=";
   };
 
+  patches = [
+    # clang no longer allows using -Ofast
+    # see: https://github.com/codam-coding-college/MLX42/issues/147
+    (fetchpatch {
+      name = "replace-ofast-with-o3.patch";
+      url = "https://github.com/codam-coding-college/MLX42/commit/ce254c3a19af8176787601a2ac3490100a5c4c61.patch";
+      hash = "sha256-urL/WVOXinf7hWR5kH+bAVTcAzldkkWfY0+diSf7jHE=";
+    })
+  ];
+
   postPatch =
     ''
       patchShebangs --build ./tools
     ''
     + lib.optionalString enableShared ''
       substituteInPlace CMakeLists.txt \
-          --replace-fail "mlx42 STATIC" "mlx42 SHARED"
+        --replace-fail "mlx42 STATIC" "mlx42 SHARED"
     '';
 
   strictDeps = true;

